docs/interactions/Slash_Commands.md

```diff
@@ -0,0 +1,841 @@
+
+# Your public key can be found on your application in the Developer Portal
+PUBLIC_KEY = 'APPLICATION_PUBLIC_KEY'
+
+verify_key = VerifyKey(bytes.fromhex(PUBLIC_KEY))
+
+signature = request.headers["X-Signature-Ed25519"]
+timestamp = request.headers["X-Signature-Timestamp"]
+body = request.data
+
+try:
+    verify_key.verify(f'{timestamp}{body}'.encode(), bytes.fromhex(signature))
+except BadSignatureError:
+    abort(401, 'invalid request signature')
+```
+
+If you are not properly validating this signature header, we will not allow you to save your interactions URL in the Dev Portal. We will also do automated, routine security checks against your endpoint, including purposefully sending you invalid signatures. If you fail the validation, we will remove your interactions URL in the future and alert you via email and System DM.
+
+We highly recommend checking out our [Community Resources](#DOCS_TOPICS_COMMUNITY_RESOURCES/interactions) and the two libraries found there. They not only provide typing for Interactions data models, but also include decorators for API frameworks like Flask and Express to make validation easy.
+
+## Subcommands and Subcommand Groups
+
+> warn
+> Currently, subcommands and subcommand groups all appear at the top level in the command explorer. This may change in the future to include them as nested autocomplete options.
+
+For those developers looking to make more organized and complex groups of commands, look no further than subcommands and groups.
+
+**Subcommands** organize your command by **specifying actions within a command or group**.
+
+**Subcommand Groups** organize your **subcommands** by **grouping commands by similar action or resource within a command**.
+
+These are not enforced rules. You are free to use subcommands and groups however you'd like; it's just how we think about them.
+
+> danger
+> Using subcommands or subcommand groups will make your base command unusable. You can't access the base `/permissions` if you also have `/permissions add | remove` as subcommands or subcommand groups
+
+### Nested Subcommands and Groups
+
+A quick note on nested subcommands and groups. We support nesting one level deep within a group, meaning your top level command can contain subcommand groups, and those groups can contain subcommands. **That is the only kind of nesting supported.** Here's some visual examples:
+
+```
+VALID
+
+command
+|
+|__ subcommand-group
+    |
+    |__ subcommand
+|
+|__ subcommand-group
+    |
+    |__ subcommand
+
+
+-------
+
+INVALID
+
+
+command
+|
+|__ subcommand-group
+    |
+    |__ subcommand-group
+|
+|__ subcommand-group
+    |
+    |__ subcommand-group
+
+----
+
+INVALID
+
+command
+|
+|__ subcommand
+    |
+    |__ subcommand-group
+|
+|__ subcommand
+    |
+    |__ subcommand-group
+```
+
+### Example Walkthrough
+
+Let's look at an example. Let's imagine you run a moderation bot. You want to make a `/permissions` command that can do the following:
```

Comment:
Let's imagine you want to know whether a user has a role called "Muted" and you don't have the id. Whoops, you cannot do that with the current system.

^ Just me trying to remind you that whenever you add smarter tokens (if smarter tokens means more perms) to also give roles or something. Given that Discord is pushing commands to be stateless, this is a requirement (especially since you might not have a bot in a guild to keep track of the roles).

Reply:
Well, a few things:

> you cannot do that with the current system.

If you still need permissions within the guild, you should still be using a bot user. I even explicitly stated that at the start of the docs.

> add smarter tokens (if smarter tokens means more perms) to also give roles or something. 

When I offhandedly mentioned smarter tokens, we're actually thinking more than this. What if the interaction token was a short-lived token scoped to the _user's_ permissions? So if I want to, say, create a channel via a bot, the bot wouldn't have to worry about its own permissions _and_ checking the user. It could just take the interactions token and hit the API and get a response without needing any permissions of its own.

^ that's all blue sky right now but an idea :)

Past this, let's talk in feature requests and comments off this PR tomorrow.